### ISSUE TITLE
How to deal with huge datasets for training NER models

### ISSUE THREAD

**Issue by frtacoa** [flairNLP/flair#671]
Hi guys,

I am trying to create a NER model using a huge dataset:
- training set: **16GB**
- validation set: 2GB
- test set: 2GB

The training set does not fit into memory, so I would like to know if there is a way to split it and use it in a similar way to [training the language models](https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_9_TRAINING_LM_EMBEDDINGS.md#preparing-a-text-corpus).

Thanks!

**Comment by alanakbik** [comment]
Hello @frtacoa you could check out [this lazy loading PR](https://github.com/zalandoresearch/flair/pull/595) by @mfojtak which implements data loaders that do not load everything into memory. It's still undergoing development so for instance it does not shuffle data, but I've tried it on large classification datasets and it works. 

We are still in the process of checking whether we should use PyTorch data loaders here, so it may be a while until this feature goes into beta on the master branch. But it is high up on our feature list (see #563) so we are planning to allocate time to this as soon as possible.

**Comment by frtacoa** [comment]
Hi @alanakbik , thanks for sharing about this!
I'll check it out.
